src/System.Data.SqlClient/src/System/Data/SqlClient/SqlParameter.cs

```diff
@@ -74,12 +74,23 @@ internal XmlDataFeed(XmlReader source)
     [TypeConverter(typeof(SqlParameterConverter))]
     public sealed partial class SqlParameter : DbParameter, IDbDataParameter, ICloneable
     {
-        private MetaType _metaType;
+        [Flags]
+        private enum SqlParameterFlags : short
+        {
+            None = 0,
+            IsNullable = 1 << 1,
+            IsNull = 1 << 2,
+            IsSqlParameterSqlType = 1 << 3,
+            CoercedValueIsSqlType = 1 << 4,
+            CoercedValueIsDataFeed = 1 << 5,
+            IsDerivedParameterTypeName = 1 << 6,
+            SourceColumnNullMapping = 1 << 7,
+            HasScale = 1 << 9,  // V1.0 compat, ignore _hasScale
```

Comment:
why not  1 << 8 ?

Reply:
Hmmm. I see two options.
* there was an 8 and i removed it, though i can't think what it might have been.
* i have an irrational and undiscovered fear of the number 8 that is causing me to avoid using it use it.

I think option 1 is more likely 😀 so i should switch it back. I'll do that tomorrow.

Reply:
Would have been funny if you had skipped 9 😊

Reply:
I've worked out why. When I started writing the fix I thought I'd need two flags and had added them in. Later I realised it was simpler than I'd thought and removed the additional flag but didn't clean up the hole. It's cleaned now.